### ISSUE TITLE
Unable to reproduce `jax.numpy.cumsum` on trn

### ISSUE THREAD

**Issue by cmrfrd** [aws-neuron/aws-neuron-sdk#767]
The follow xla hlo when run on trn does not produce the same output when run via `jax`.

JAX code:

```python
import jax
import jax.numpy as jnp
import jax.lax as lax

N = 10

def f():
    a = jnp.arange(N)
    return jnp.cumsum(a, axis=0)
comp = jax.xla_computation(f)()
print(comp.as_hlo_text())
open("hlo.pb", "wb").write(comp.as_serialized_hlo_module_proto())
print("out: ", f())
```
```
out:  [ 0  1  3  6 10 15 21 28 36 45]
```

XLA hlo
```
HloModule xla_computation_f, entry_computation_layout={()->(s32[10]{0})}

cumsum.2 {
  Arg_0.3 = s32[10]{0} parameter(0)
  slice.28 = s32[1]{0} slice(Arg_0.3), slice={[0:1]}
  slice.5 = s32[5]{0} slice(Arg_0.3), slice={[0:9:2]}
  slice.6 = s32[5]{0} slice(Arg_0.3), slice={[1:10:2]}
  add.7 = s32[5]{0} add(slice.5, slice.6)
  slice.20 = s32[1]{0} slice(add.7), slice={[0:1]}
  slice.8 = s32[2]{0} slice(add.7), slice={[0:4:2]}
  slice.9 = s32[2]{0} slice(add.7), slice={[1:5:2]}
  add.10 = s32[2]{0} add(slice.8, slice.9)
  slice.14 = s32[1]{0} slice(add.10), slice={[0:1]}
  constant.4 = s32[] constant(0)
  pad.15 = s32[2]{0} pad(slice.14, constant.4), padding=0_1_1
  slice.11 = s32[1]{0} slice(add.10), slice={[0:1:2]}
  slice.12 = s32[1]{0} slice(add.10), slice={[1:2:2]}
  add.13 = s32[1]{0} add(slice.11, slice.12)
  pad.16 = s32[2]{0} pad(add.13, constant.4), padding=1_0_1
  add.17 = s32[2]{0} add(pad.15, pad.16)
  slice.18 = s32[2]{0} slice(add.7), slice={[2:5:2]}
  add.19 = s32[2]{0} add(add.17, slice.18)
  concatenate.21 = s32[3]{0} concatenate(slice.20, add.19), dimensions={0}
  pad.22 = s32[5]{0} pad(concatenate.21, constant.4), padding=0_0_1
  pad.23 = s32[5]{0} pad(add.17, constant.4), padding=1_1_1
  add.24 = s32[5]{0} add(pad.22, pad.23)
  slice.25 = s32[4]{0} slice(add.24), slice={[0:4]}
  slice.26 = s32[4]{0} slice(Arg_0.3), slice={[2:10:2]}
  add.27 = s32[4]{0} add(slice.25, slice.26)
  concatenate.29 = s32[5]{0} concatenate(slice.28, add.27), dimensions={0}
  pad.30 = s32[10]{0} pad(concatenate.29, constant.4), padding=0_1_1
  pad.31 = s32[10]{0} pad(add.24, constant.4), padding=1_0_1
  ROOT add.32 = s32[10]{0} add(pad.30, pad.31)
}

_cumulative_reduction.33 {
  Arg_0.34 = s32[10]{0} parameter(0)
  ROOT call.35 = s32[10]{0} call(Arg_0.34), to_apply=cumsum.2
}

ENTRY main.38 {
  iota.1 = s32[10]{0} iota(), iota_dimension=0
  call.36 = s32[10]{0} call(iota.1), to_apply=_cumulative_reduction.33
  ROOT tuple.37 = (s32[10]{0}) tuple(call.36)
}
```
[hlo.pb.zip](https://github.com/aws-neuron/aws-neuron-sdk/files/12898307/hlo.pb.zip)

When running this hlo on trn via our rust framework:

```rust
        let mut out_tensor: Tensor<i32> = Tensor::new_op(&[10], &[1], false, None, None);
        let outputs = vec![TensorData::I32(out_tensor.clone())];
        let out_map: HashMap<String, TensorData> = outputs
            .iter()
            .enumerate()
            .map(|(i, tensor)| (format!("{}{}", "output", i), tensor.clone()))
            .collect();

        let conf = &ProcessedXlaRunConfig {
            xla_hlo_pb_path: "./hlo.pb".to_string(),
            run_name: "cumsum".to_string(),
            inputs: HashMap::new(),
            outputs: out_map,
            benchmark: false,
            model: None,
        };
        RUNNER.run_trn(conf).unwrap();
        let out = out_tensor.read_values().borrow().clone();
        println!("{:?}", out);
```

this is our output.
```
[0, 0, 2, 6, 10, 15, 21, 28, 36, 45]
```

This is the version of the compiler being used:

```
$ neuronx-cc --version
NeuronX Compiler version 2.10.0.35+3817a0c8c

Python version 3.8.10
HWM version 2.10.0.5-7b1976adf
NumPy version 1.21.6
```

**Comment by cmrfrd** [comment]
Additionally doing a `cumsum` like this `jnp.cumsum(jnp.arange(2, dtype=jnp.int32), axis=0)` causes a compiler crash.

hlo:
```
compiled debug_ir: HloModule xla_computation_f, entry_computation_layout={()->(s32[2]{0})}

cumsum.2 {
  Arg_0.3 = s32[2]{0} parameter(0)
  slice.8 = s32[1]{0} slice(Arg_0.3), slice={[0:1]}
  constant.4 = s32[] constant(0)
  pad.9 = s32[2]{0} pad(slice.8, constant.4), padding=0_1_1
  slice.5 = s32[1]{0} slice(Arg_0.3), slice={[0:1:2]}
  slice.6 = s32[1]{0} slice(Arg_0.3), slice={[1:2:2]}
  add.7 = s32[1]{0} add(slice.5, slice.6)
  pad.10 = s32[2]{0} pad(add.7, constant.4), padding=1_0_1
  ROOT add.11 = s32[2]{0} add(pad.9, pad.10)
}

_cumulative_reduction.12 {
  Arg_0.13 = s32[2]{0} parameter(0)
  ROOT call.14 = s32[2]{0} call(Arg_0.13), to_apply=cumsum.2
}

ENTRY main.17 {
  iota.1 = s32[2]{0} iota(), iota_dimension=0
  call.15 = s32[2]{0} call(iota.1), to_apply=_cumulative_reduction.12
  ROOT tuple.16 = (s32[2]{0}) tuple(call.15)
}
```
[hlo.pb.zip](https://github.com/aws-neuron/aws-neuron-sdk/files/12911675/hlo.pb.zip)

crash logs:

```
...
2023-10-16T03:01:00Z ERROR 1703836 140286315059008 [neuronxcc.driver.CommandDriver]: ***************************************************************
2023-10-16T03:01:00Z ERROR 1703836 140286315059008 [neuronxcc.driver.CommandDriver]:  An Internal Compiler Error has occurred
2023-10-16T03:01:00Z ERROR 1703836 140286315059008 [neuronxcc.driver.CommandDriver]: ***************************************************************
2023-10-16T03:01:00Z ERROR 1703836 140286315059008 [neuronxcc.driver.CommandDriver]: 
2023-10-16T03:01:00Z USER 1703836 140286315059008 [neuronxcc.driver.CommandDriver]: [F134] neuronx-cc terminated abnormally - Please open a support ticket at https://github.com/aws-neuron/aws-neuron-sdk/issues/new
2023-10-16T03:01:00Z ERROR 1703836 140286315059008 [neuronxcc.driver.CommandDriver]: 
2023-10-16T03:01:00Z ERROR 1703836 140286315059008 [neuronxcc.driver.CommandDriver]: Internal details:
2023-10-16T03:01:00Z ERROR 1703836 140286315059008 [neuronxcc.driver.CommandDriver]: Type: <class 'RuntimeError'>
2023-10-16T03:01:00Z ERROR 1703836 140286315059008 [neuronxcc.driver.CommandDriver]:   File "neuronxcc/driver/Job.py", line 215, in neuronxcc.driver.Job.Job.shellCommand
2023-10-16T03:01:00Z USER 1703836 140286315059008 [neuronxcc.driver.CommandDriver]: 
2023-10-16T03:01:00Z USER 1703836 140286315059008 [neuronxcc.driver.CommandDriver]: Diagnostic information:
2023-10-16T03:01:00Z USER 1703836 140286315059008 [neuronxcc.driver.CommandDriver]:   NeuronX Compiler version 2.10.0.35+3817a0c8c
2023-10-16T03:01:00Z USER 1703836 140286315059008 [neuronxcc.driver.CommandDriver]:   
2023-10-16T03:01:00Z USER 1703836 140286315059008 [neuronxcc.driver.CommandDriver]:   Python version 3.8.10
2023-10-16T03:01:00Z USER 1703836 140286315059008 [neuronxcc.driver.CommandDriver]:   HWM version 2.10.0.5-7b1976adf
2023-10-16T03:01:00Z USER 1703836 140286315059008 [neuronxcc.driver.CommandDriver]:   NumPy version 1.21.6
2023-10-16T03:01:00Z USER 1703836 140286315059008 [neuronxcc.driver.CommandDriver]:   
2023-10-16T03:01:00Z USER 1703836 140286315059008 [neuronxcc.driver.CommandDriver]:   Running on AMI ami-0d6b6f1008335d1d5
2023-10-16T03:01:00Z USER 1703836 140286315059008 [neuronxcc.driver.CommandDriver]:   Running in region usw2-az4
2023-10-16T03:01:00Z USER 1703836 140286315059008 [neuronxcc.driver.CommandDriver]: 
2023-10-16T03:01:00Z USER 1703836 140286315059008 [neuronxcc.driver.CommandDriver]: Diagnostic logs stored in /home/ubuntu/dev/Kholinar/xla/log-neuron-cc.txt
2023-10-16T03:01:00Z INFO 1703836 140286315059008 [neuronxcc.driver.CommandDriver]: Artifacts stored in: /home/ubuntu/dev/Kholinar/xla/neuronxcc-snxnvip4
...
```
[log-neuron-cc.txt](https://github.com/aws-neuron/aws-neuron-sdk/files/12911679/log-neuron-cc.txt)

**Comment by aws-donkrets** [comment]
Hi [cmrfrd](https://github.com/cmrfrd), quick update - we were able to reproduce your reported error and are working on a fix.

**Comment by aws-donkrets** [comment]
Hi [cmrfrd](https://github.com/cmrfrd), what was the error? Since the compilation completed and execution was successful, could it simply be a warning? Can we close this ticket?

**Comment by cmrfrd** [comment]
Yes we can close this ticket, thank you !
